pandas/core/common.py

```diff
@@ -1403,6 +1403,10 @@ def _fill_zeros(result, x, y, name, fill):
                 if np.isinf(fill):
                     np.putmask(result,(signs<0) & mask, -fill)
 
+                if "floordiv" in name:  # (PR 9308)
+                    nan_mask = ((y == 0) & (x == 0)).ravel()
+                    np.putmask(result, nan_mask, np.nan)
+
```

Comment:
This is pretty surprising I suppose.  I even tried experimenting with having both the unraveled version and raveled version together.  Here's the pertinent parts of the line profiling:

    Line #      Hits         Time  Per Hit   % Time  Line Contents
    ==============================================================

      1405         5      1715640 343128.0      9.9          result_rav = result.ravel()
      1406         5           29      5.8      0.0          mask_rav = mask.ravel()
      1407                                           
      1408         5      3943703 788740.6     22.7          np.putmask(result, mask, fill)
      1409                                           
      1410         5       443946  88789.2      2.6          np.putmask(result_rav, mask_rav, fill)